parse.go

```diff
@@ -0,0 +1,566 @@
+package dicom
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/gradienthealth/go-dicom/dicomio"
+	"github.com/gradienthealth/go-dicom/dicomlog"
+	"github.com/gradienthealth/go-dicom/dicomtag"
+)
+
+// GoDICOMImplementationClassUIDPrefix defines the UID prefix for
+// go-dicom. Provided by https://www.medicalconnections.co.uk/Free_UID
+const GoDICOMImplementationClassUIDPrefix = "1.2.826.0.1.3680043.9.7133"
+
+var GoDICOMImplementationClassUID = GoDICOMImplementationClassUIDPrefix + ".1.1"
+
+const GoDICOMImplementationVersionName = "GODICOM_1_1"
+
+// Parser represents an entity that can read and parse DICOMs
+type Parser interface {
```

Comment:
new interface! :) 
(github keeps collapsing this file because it thinks it's too large of a diff)

Reply:
was thinking maybe `dicom.Reader` would be cool, but Parser is maybe more clear? would be nice to pair with `dicom.Writer` for the writing func in this lib